test/InjectorTest.php

```diff
@@ -341,4 +343,127 @@ public function testComplexDeepDependencyConfiguration()
         $this->assertSame($expected1, $result->result->result->optionalResult);
         $this->assertSame($expected2, $result->result2->result->optionalResult);
     }
+
+    public function testCreateInstanceWithoutUnknownClassThrowsException()
+    {
+        $this->expectException(Exception\ClassNotFoundException::class);
+        (new Injector())->create('Unknown.Alias.Should.Fail');
+    }
+
+    public function testKnownButInexistentClassThrowsException()
+    {
+        $definition = $this->getMockBuilder(DefinitionInterface::class)
+                           ->getMockForAbstractClass();
+
+        $definition->expects($this->any())
+                   ->method('hasClass')
+                   ->willReturn(true);
+
+        $this->expectException(Exception\ClassNotFoundException::class);
+        (new Injector(null, null, $definition))->create('ZendTest\Di\TestAsset\No\Such\Class');
+    }
+
+    public function provideUnexpectedResolverValues()
+    {
+        return [
+            [ 'string value' ],
+            [ true ],
+            [ null ],
+            [ new \stdClass() ]
```

Comment:
Two notes:

- Provide names to each case; e.g., `'string' => ['string value'],`. This simplifies tracking down _which_ case failed, and also allows us to use `phpunit --filter` with the name to run single tests.
- No extra spaces within the arrays, please.